### PR TITLE
Update MDK to 1.20.4

### PR DESCRIPTION
**Pull request by Matyrobbrt**
Document the new `type` and `reason` fields

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -6,17 +6,17 @@ org.gradle.debug=false
...
+minecraft_version=1.20.4
 # The Minecraft version range can use any release version of Minecraft as bounds.
 # Snapshots, pre-releases, and release candidates are not guaranteed to sort properly
 # as they do not follow standard versioning conventions.
-minecraft_version_range=[1.20.2,1.21)
+minecraft_version_range=[1.20.4,1.21)
 # The Neo version must agree with the Minecraft version to get a valid artifact
-neo_version=20.2.86
+neo_version=20.4.42-beta
 # The Neo version range can use any version of Neo as bounds or match the loader version range
```
Comment (Technici4n) [inline gradle.properties:16]: should we update this comment and the one below as well?

File: gradle.properties [hunk 2]
```diff
@@ -38,4 +38,4 @@ mod_authors=YourNameHere, OtherNameHere
 # The description of the mod. This is a simple multiline text string that is used for display purposes in the mod list.
 mod_description=Example mod description.\nNewline characters can be used and will be replaced properly.
 # Pack version - this changes each minecraft release, in general.
```
Comment (Technici4n) [inline gradle.properties:40]: Can't we use the new version range system? (See NeoForge itself)

Comment (Matyrobbrt) [reply gradle.properties:40]: Honestly, I'd rather we no longer require mods to provide any pack version information, and we just provide an all-inclusive compatibility by default.

**Comment by Technici4n** [comment]
Update?

**Pull request merged**